packages/blocks/src/api/utils.js

```diff
@@ -264,9 +279,28 @@ export function __experimentalSanitizeBlockAttributes( name, attributes ) {
 			const value = attributes[ key ];
 
 			if ( undefined !== value ) {
-				accumulator[ key ] = value;
-			} else if ( schema.hasOwnProperty( 'default' ) ) {
-				accumulator[ key ] = schema.default;
+				if ( schema.type === 'rich-text' ) {
+					if ( value instanceof RichTextData ) {
+						accumulator[ key ] = value;
+					} else if ( typeof value === 'string' ) {
+						accumulator[ key ] = new RichTextData( {
+							html: value,
+						} );
+					}
+				} else if ( schema.type === 'string' ) {
+					if ( typeof value === 'string' ) {
+						accumulator[ key ] = value;
+					} else if ( value instanceof RichTextData ) {
+						accumulator[ key ] = value.toString();
+					}
+				} else {
+					accumulator[ key ] = value;
```

Comment:
It's quite strange to me that we don't enforce types. You can do this currently in trunk:

```
wp.blocks.createBlock( 'core/heading', { level: 'string' } )
```